float64 visual
float64 stored
---
string res
